onnxruntime/core/session/allocator_adapters.cc

```diff
@@ -26,6 +27,21 @@ OrtAllocatorImplWrappingIAllocator::OrtAllocatorImplWrappingIAllocator(onnxrunti
     OrtAllocator::Reserve =
         [](OrtAllocator* this_, size_t size) { return static_cast<OrtAllocatorImplWrappingIAllocator*>(this_)->Reserve(size); };
   }
+  if (OrtAllocator::version >= kOrtAllocatorStatsMinVersion) {
+    OrtAllocator::GetStats =
+        [](const OrtAllocator* this_, OrtAllocator* allocator, char** stats) {
+          auto str = static_cast<const OrtAllocatorImplWrappingIAllocator*>(this_)->Stats();
```

Comment:
>auto str = static_cast<const OrtAllocatorImplWrappingIAllocator*>(this_)->Stats(); [](http://example.com/codeflow?start=10&length=82)

Handle the exception here and return as OrtStatus, use the utility function to perform the conversion or use the API macro that does it automatically